skyportal/handlers/api/photometry.py

```diff
@@ -41,46 +34,59 @@ def post(self):
-                              '(\'time_format\') are required parameters.')
-        if not isinstance(data['mag'], (list, tuple)):
-            data['observed_at'] = [data['observed_at']]
-            data['mag'] = [data['mag']]
-            data['e_mag'] = [data['e_mag']]
-            data['lim_mag'] = [data['lim_mag']]
-            data['filter'] = [data['filter']]
+        if "time_format" not in data or "time_scale" not in data:
+            return self.error(
+                "Time scale ('time_scale') and time format "
+                "('time_format') are required parameters."
+            )
+        if not isinstance(data["mag"], (list, tuple)):
+            data["observed_at"] = [data["observed_at"]]
+            data["mag"] = [data["mag"]]
+            data["e_mag"] = [data["e_mag"]]
+            data["lim_mag"] = [data["lim_mag"]]
+            data["filter"] = [data["filter"]]
         ids = []
-        instrument = Instrument.query.get(data['instrument_id'])
+        instrument = Instrument.query.get(data["instrument_id"])
         if not instrument:
-            raise Exception('Invalid instrument ID') # TODO: handle invalid instrument ID
-        source = Source.get_if_owned_by(data['source_id'], self.current_user)
-        if not source:
-            raise Exception('Invalid source ID') # TODO: handle invalid source ID
-        for i in range(len(data['mag'])):
-            if not (data['time_scale'] == 'tcb' and data['time_format'] == 'iso'):
-                t = Time(data['observed_at'][i],
-                         format=data['time_format'],
-                         scale=data['time_scale'])
-                observed_at = t.tcb.iso
-            else:
-                observed_at = data['time'][i]
-            p = Photometry(source=source,
-                           observed_at=observed_at,
-                           mag=data['mag'][i],
-                           e_mag=data['e_mag'][i],
-                           time_scale='tcb',
-                           time_format='iso',
-                           instrument=instrument,
-                           lim_mag=data['lim_mag'][i],
-                           filter=data['filter'][i])
+            return self.error("Invalid instrument ID")
+        source_id = data.get("source_id", None)
+        source = get_either_source_candidate_if_owned_by_user(source_id, self.current_user)
+        converted_times = []
+        for i in range(len(data["mag"])):
+            t = Time(
+                data["observed_at"][i].replace("T", " ").split("+")[0],
+                format=data["time_format"],
+                scale=data["time_scale"],
+            )
+            observed_at = arrow.get(t.tcb.iso)
+            converted_times.append(observed_at)
+            p = Photometry(
+                source_id=source_id,
+                observed_at=observed_at,
+                mag=data["mag"][i],
+                e_mag=data["e_mag"][i],
+                time_scale="tcb",
```

Comment:
these will be removed in #382 I believe.

Reply:
We still have these fields in that PR - perhaps a future PR will need to update this